fix type for argument to h5dget_type_f call
see https://github.com/EDmodel/ED2/issues/153 and https://github.com/crollinson/ED2/commit/48b9e8132b992460aa8afeca222fd4c12f01cfe4
--- ED/src/utils/hdf5_utils.F90.orig	2017-02-01 22:00:01.000000000 +0100
+++ ED/src/utils/hdf5_utils.F90	2017-03-27 11:56:38.579744327 +0200
@@ -432,7 +432,7 @@
   character(len=2) :: ctype
   
   logical :: convert = .false.
-  integer :: type_id
+  integer(HID_T) :: type_id
   real(kind=8), allocatable, dimension(:) :: dvaraTEMP
   
   ! Find which data type will be read
